source/Calamari.Common/Plumbing/Deployment/PackageRetention/IJournal.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using Calamari.Common.Plumbing.Variables;
+using Octopus.Versioning;
+
+namespace Calamari.Common.Plumbing.Deployment.PackageRetention
+{
+    public interface IJournal
+    {
+        void RegisterPackageUse(string packageID, string version, string serverTaskID);
```

Comment:
This signature with string parameters doesn't look used, and I think the `PackageIdentity` signatures are probably a better option to let the compiler kick in sooner. Can axe?

Reply:
💯